mesonbuild/compilers/cuda.py

```diff
@@ -233,7 +235,7 @@ def _shield_nvcc_list_arg(cls, arg: str, listmode: bool = True) -> str:
                 # There are single quotes. Double-quote them, and single-quote the
                 # strings between them.
                 l = [cls._shield_nvcc_list_arg(s) for s in arg.split(SQ)]
-                l = sum([[s, DQSQ] for s in l][:-1], [])  # Interleave l with DQSQs
+                l = functools.reduce(operator.iadd, [[s, DQSQ] for s in l][:-1], [])  # Interleave l with DQSQs
                 return ''.join(l)
```

Comment:
This whole logic feels kinda hairy honestly. I wonder if there's a better way to do it that doesn't need additional imports...

Reply:
Hairy indeed.  The commit message lists the three linear solutions and why ruff defaults to this one.  Given it hairiness, let's revert this change and deal with it in a separate PR.